### PR TITLE
MGMT-4087 - add REST API for static network configuration

### PR DESCRIPTION
**Pull request by yevgeny-shnaidman**

### CODE REVIEW

File: swagger.yaml [hunk 1]
```diff
@@ -2944,18 +2944,14 @@ definitions:
       ssh_public_key:
         type: string
         description: SSH public key for debugging the installation.
-      static_ips_config:
+      static-network_config:
```
Comment (avishayt) [inline swagger.yaml:2947]:
```suggestion
      static_network_config:
```

File: swagger.yaml [hunk 2]
```diff
@@ -2944,18 +2944,14 @@ definitions:
...
         type: array
         items:
-          $ref: '#/definitions/static-ip-config'
+          type: string
```
Comment (avishayt) [inline swagger.yaml:2950]: Please add comment about nmstate format

Comment (yevgeny-shnaidman) [reply swagger.yaml:2950]: done

File: swagger.yaml (hunk 1 above)
Comment (yevgeny-shnaidman) [reply swagger.yaml]: done

File: internal/bminventory/inventory.go [hunk 3]
```diff
@@ -431,10 +430,7 @@ func (b *bareMetalInventory) formatIgnitionFile(cluster *common.Cluster, params
 		ignitionParams["ServiceIPs"] = dataurl.EncodeBytes([]byte(ignition.GetServiceIPHostnames(b.Config.ServiceIPs)))
 	}
 
-	if cluster.ImageInfo.StaticIpsConfig != "" && params.ImageCreateParams.ImageType == models.ImageTypeFullIso {
-		ignitionParams["StaticIPsData"] = base64.StdEncoding.EncodeToString([]byte(cluster.ImageInfo.StaticIpsConfig))
-		ignitionParams["StaticIPsConfigScript"] = base64.StdEncoding.EncodeToString([]byte(constants.ConfigStaticIpsScript))
-	}
+	// [TODO] - add static network configuration only in case it was provided and this is a FullIso image
```
Comment (filanov) [inline internal/bminventory/inventory.go:433]: Will it be done in a different PR?

File: internal/bminventory/inventory.go [hunk 4]
```diff
@@ -1062,15 +1058,12 @@ func (b *bareMetalInventory) GenerateClusterISOInternal(ctx context.Context, par
...
-
-	vlansConfig := formatVlans(params.ImageCreateParams.VlansConfig)
+	staticNetworkConfig := formatStaticNetwork(params.ImageCreateParams.StaticNetworkConfig)
 
 	var imageExists bool
 	if cluster.ImageInfo.SSHPublicKey == params.ImageCreateParams.SSHPublicKey &&
 		cluster.ProxyHash == clusterProxyHash &&
-		cluster.ImageInfo.StaticIpsConfig == staticIpsConfig &&
-		cluster.ImageInfo.VlansConfig == vlansConfig &&
+		cluster.ImageInfo.StaticNetworkConfig == staticNetworkConfig &&
```
Comment (filanov) [inline internal/bminventory/inventory.go:1066]: Isn't it breaking existing API?

File: internal/bminventory/inventory.go [hunk 5]
```diff
@@ -4451,42 +4444,12 @@ func (b *bareMetalInventory) setPullSecretFromOCP(cluster *common.Cluster, log l
...
-
-	return strings.Join(lines, "\n")
-}
-
-func getStaticIPsData(static_config *models.StaticIPV4Config) []string {
-	if static_config != nil {
-		return []string{static_config.IP, static_config.Mask, static_config.DNS, static_config.Gateway}
-	}
-	return []string{"", "", "", ""}
+	return strings.Join(lines, "ZZZZZ")
```
Comment (filanov) [inline internal/bminventory/inventory.go:4452]: what is `ZZZZZ`

File: internal/bminventory/inventory_test.go [hunk 6]
```diff
@@ -327,192 +327,195 @@ var _ = Describe("GenerateClusterISO", func() {
 		verifyApiError(generateReply, http.StatusInternalServerError)
 	})
 
-	Context("static ip and vlan", func() {
+	// [TODO] change the tests once nmstate-based implementation is in
```
Comment (filanov) [inline internal/bminventory/inventory_test.go:330]: When is it going to be fixed?

File: internal/bminventory/inventory.go (hunk 3 above)
Comment (yevgeny-shnaidman) [reply internal/bminventory/inventory.go:433]: yes, all part of https://issues.redhat.com/browse/MGMT-3867

File: internal/bminventory/inventory.go (hunk 4 above)
Comment (yevgeny-shnaidman) [reply internal/bminventory/inventory.go:1066]: it does. we are deleting the current API, and the PR for disabling the e2e and test-infra test are already pushed

File: internal/bminventory/inventory.go (hunk 5 above)
Comment (yevgeny-shnaidman) [reply internal/bminventory/inventory.go:4452]: i needed some delimeter between hosts configuration. ZZZZ sounds as good as any

File: internal/bminventory/inventory_test.go (hunk 6 above)
Comment (yevgeny-shnaidman) [reply internal/bminventory/inventory_test.go:330]: i am already working on the PR

File: internal/bminventory/inventory.go (hunk 5 above)
Comment (filanov) [reply internal/bminventory/inventory.go:4452]: then please add a comment explaining it.

Comment (yevgeny-shnaidman) [reply internal/bminventory/inventory.go:4443]: done

**Review by avishayt — APPROVED** [review]

**Review by avishayt — APPROVED** [review]
